### ISSUE TITLE
whitespace rule 'check-decl' fails when statement does not include a variable declaration keyword

### ISSUE THREAD

**Issue by robert-prib-polestar** [palantir/tslint#3199]
### Bug Report

- __TSLint version__: 3.10.10
- __TypeScript version__: 3.10.10
- __Running TSLint via__:  VSCode 

#### TypeScript code being linted

```ts

let x= 1;// displays warning
x= 0;// displays warning
class test {

 private x= 1;// displays warning
 y= 1;//does not display warning
 public c=0;//does not display warning
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "member-access": false,
    "member-ordering": [
      false,
      "public-before-private",
      "static-before-instance",
      "variables-before-functions"
    ],
    "no-any": false,
    "no-inferrable-types": false,
    "no-internal-module": true,
    "no-var-requires": false,
    "typedef": false,
    "ban": false,
    "curly": false,
    "forin": true,
    "label-position": true,
    "label-undefined": true,
    "no-arg": true,
    "no-bitwise": false,
    "no-conditional-assignment": true,
    "no-console": [
      true,
      "debug",
      "info",
      "time",
      "timeEnd",
      "trace"
    ],
    "no-construct": true,
    "no-debugger": true,
    "no-duplicate-key": true,
    "no-duplicate-variable": true,
    "no-empty": false,
    "no-eval": true,
    "no-null-keyword": true,
    "no-shadowed-variable": false,
    "no-string-literal": false,
    "no-switch-case-fall-through": true,
    "no-trailing-whitespace": true,
    "no-unreachable": true,
    "no-unused-expression": true,
    "no-unused-variable": true,
    "no-use-before-declare": true,
    "no-var-keyword": true,
    "radix": true,
    "switch-default": true,
    "triple-equals": [
      true,
      "allow-null-check"
    ],
    "typedef-whitespace": [
      true,
      {
        "call-signature": "nospace",
        "index-signature": "nospace",
        "parameter": "nospace",
        "property-declaration": "nospace",
        "variable-declaration": "nospace"
      },
      {
        "call-signature": "onespace",
        "index-signature": "onespace",
        "parameter": "onespace",
        "property-declaration": "onespace",
        "variable-declaration": "onespace"
      }
    ],
    "use-strict": [
      true,
      "check-module"
    ],

    "eofline": true,
    "indent": [
      true,
      "spaces"
    ],
    "max-line-length": [
      true,
      130
    ],
    "no-require-imports": false,
    "object-literal-sort-keys": false,
    "trailing-comma": [
      true,
      {
        "multiline": "never",
        "singleline": "never"
      }
    ],

    "align": false,
    "class-name": true,
    "comment-format": [
      true,
      "check-space"
    ],
    "interface-name": false,
    "jsdoc-format": true,
    "no-consecutive-blank-lines": false,
    "no-constructor-vars": false,
    "one-line": [
      true,
      "check-open-brace",
      "check-catch",
      "check-else",
      "check-finally",
      "check-whitespace"
    ],
    "quotemark": [
      true,
      "single",
      "avoid-escape"
    ],
    "semicolon": [true, "always"],
    "variable-name": [
      true,
      "allow-leading-underscore",
      "ban-keywords"
    ],
    "whitespace": [
      true,
      "check-branch",
      "check-decl",
      "check-operator",
      "check-separator",
      "check-type"
    ]

  }
}

```

#### Actual behavior
No white space before '=' operators in assignment declarations within a class does not display a warning.

#### Expected behaviour
The warning is displayed for no white space before '=' operator in a variable declaration.

**Comment by ChaseMoskal** [comment]
As shown in #3344, it appears that `check-decl` may also be malfunctioning in some cases **when the variable declaration keyword is present.**

**Comment by ajafff** [comment]
Fixed by #3546
